src/main/web/CMakeLists.txt

```diff
@@ -2,21 +2,26 @@ set(SCRIPT "${PROJECT_SOURCE_DIR}/scripts/gen_embedded.py")
 
 
 set(EMBEDDED_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/_404.png
-    ${CMAKE_CURRENT_SOURCE_DIR}/_error.css
-    ${CMAKE_CURRENT_SOURCE_DIR}/_error.html
-    ${CMAKE_CURRENT_SOURCE_DIR}/favicon.ico
-    ${CMAKE_CURRENT_SOURCE_DIR}/_jquery.min.js
-    ${CMAKE_CURRENT_SOURCE_DIR}/_seasocks.css
-    ${CMAKE_CURRENT_SOURCE_DIR}/_stats.html
-)
+        ${CMAKE_CURRENT_SOURCE_DIR}/_404.png
+        ${CMAKE_CURRENT_SOURCE_DIR}/_error.css
+        ${CMAKE_CURRENT_SOURCE_DIR}/_error.html
+        ${CMAKE_CURRENT_SOURCE_DIR}/favicon.ico
+        ${CMAKE_CURRENT_SOURCE_DIR}/_jquery.min.js
+        ${CMAKE_CURRENT_SOURCE_DIR}/_seasocks.css
+        ${CMAKE_CURRENT_SOURCE_DIR}/_stats.html
+        )
 
 
 add_custom_command(OUTPUT Embedded.cpp
-    COMMAND "${PYTHON_EXECUTABLE}" "${SCRIPT}" -o Embedded.cpp -f ${EMBEDDED_FILES}
-    COMMENT "Generating embedded content"
-    DEPENDS "${SCRIPT}"
-)
+        COMMAND "${PYTHON_EXECUTABLE}" "${SCRIPT}" -o Embedded.cpp -f ${EMBEDDED_FILES}
+        COMMENT "Generating embedded content"
+        DEPENDS "${SCRIPT}"
+        )
 add_library(embedded OBJECT Embedded.cpp)
 target_include_directories(embedded PUBLIC $<TARGET_PROPERTY:seasocks_headers,INTERFACE_INCLUDE_DIRECTORIES>)
+# TODO; for the life of me I don't understand why this if() doesn't work and we have to always build PIC.
```

Comment:
This is the other spot I need help from @madebr : I don't understand why this has to be unconditionally PIC: I was imagining this would "just work" based on the SEASOCKS_SHARED. but it doesn't and when I look at the CMakeCache.txt in the conan build (where I test) there are confusing discrepancies between SEASOCKS_SHARED and BUILD_SHARED_LIBS)